simulator/examples/clock.rs

```diff
@@ -0,0 +1,182 @@
+//! Demonstrate usage of primitives like `fill.rs` but use macros instead for shorter code
+//!
+//! # Development findings
+//!
+//! Some of these should probably be turned into issues and discussed before this example is
+//! released.
+//!
+//! 1. The primitives macros should accept both `()` tuples and variables/expressions for their
+//! coordinate positions. In this code, I often find myself replacing a macro with a call to
+//! `[Primitive]::new()` so I can pass in a `Point`
+//! 2. I often want a relative offset from a `Point`. Maybe we implement `Transform` for `Point`? Or
+//! a new trait/set of methods to do relative moves.
```

Comment:
How would this be different than adding 2 points?

Reply:
... because I should've gone to bed long before I wrote this :wink:. I've removed this comment and just started adding/subbing `Point`s and `Size`s.